src/Console.cc

```diff
@@ -122,11 +122,16 @@ Logger::Buffer::~Buffer()
 /////////////////////////////////////////////////
 int Logger::Buffer::sync()
 {
+  static std::mutex syncMutex;
+
   std::string outstr = this->str();
 
   // Log messages to disk
-  Console::log << outstr;
-  Console::log.flush();
+  {
+    std::lock_guard<std::mutex> lk(syncMutex);
```

Comment:
Did you consider having separate mutexes for logging to disk and logging to the console? This way different loggers could write to disk and to console at the same time. It may be a small optimization, but when logging a lot it could make a difference?